craft.lic

```diff
@@ -401,10 +401,38 @@ class Craft
       DRCT.walk_to(@training_room)
       DRC.wait_for_script_to_complete('enchant',[6,'lay ward runestone','runestone'])
       DRCI.dispose_trash('runestone', @worn_trashcan, @worn_trashcan_verb)
-    elsif rank <= 1175 # Tier 10 - Difficult
-      echo('*** NOT YET IMPLEMENTED ***')
-    else # Tier 11  Very Difficult
-      echo('*** NOT YET IMPLEMENTED ***')
+    elsif rank <= 1175 # Tier 10 - Difficult - Need to use sigilhunter or similar to gather the sigils
+      # Examples for sigilhunter:
+      # ;sigilharvest crossing evolution 15 2
+      # ;sigilharvest crossing metamorphosis 15 2
+      # ;sigilharvest crossing unity 15 2
+      # or: ;multi 3,;sigilharvest crossing evolution 15 2,;sigilharvest crossing metamorphosis 15 2,;sigilharvest crossing unity 15 2
+      DRCM.ensure_copper_on_hand(4000, @settings)
+      exit if !DRCI.exists?("evolution sigil-scroll")
+      exit if !DRCI.exists?("metamorphosis sigil-scroll")
+      exit if !DRCI.exists?("unity sigil-scroll")
```

Comment:
We seem to have a fair amount of `exists?` checks here. Can we break this out into a method that helps us check?

Also, do we need some cleanup here in case we have tools in our hands or what have you? Or is a brute-force exit here appropriate?

Reply:
You can only harvest these sigils. 

> Shouldn't be any tools in our hand at this point(re cleanup)
> 
> I think the exit is appropriate because we for sure don't want to START crafting without having everything on hand for fear of wasting a sigil we have to farm, but moving it into a separate method for these checks is probably a good idea. Reason being, you have information there that the user might want(the commented information specifically), and if the script just dies, they might know they're missing one of the sigils, but not the whole list. What about sending the needed parts to a method that checks for them, and if any fail, it spits out that helpful set of comments you have in the code?
> 
> Something like this in place of the tap checks:
> 
> ```ruby
> sigils_needed = ['evolution', 'metamorphosis', 'unity']
> check_sigils(sigils_needed)
> ```
> 
> method could be something like:
> 
> ```ruby
> def check_sigils(sigils)
>   missing = []
>   missing << sigils.reject { |sigil| DRCI.exists?("#{sigil} sigil-scroll") }
>   unless missing.empty?
>     DRC.message("Missing #{missing.to_s} sigil-scroll(s). IF you have them, they need to be loose in your bag, if you do not, these cannot be purchased, and must be harvested. We recommend using our sigilhunter script. Run ;sigilhunter help for more information.")
>     exit
>   end
> end
> ```

I am working on a `common-crafting` way. I have it working with `workorders` and can use here (my plan was for that all along).